x-pack/legacy/plugins/index_management/public/components/mappings_editor/lib/mappings_validator.ts

```diff
@@ -0,0 +1,311 @@
+  if (
+    !validateFieldType(
+      field.type !== undefined && field.type !== null ? field.type : DEFAULT_FIELD_TYPE
+    )
+  ) {
+    return { parametersRemoved: [] };
+  }
+
+  // Filter out unknown or invalid "parameters"
+  const fieldWithType = { type: DEFAULT_FIELD_TYPE, ...field };
+  const parsedField = stripUnknownOrInvalidParameter(fieldWithType);
+  const meta = getFieldMeta(fieldWithType);
+
+  return { ...parsedField, meta };
+};
+
+const parseFields = (
+  properties: GenericObject,
+  path: string[] = []
+): PropertiesValidatorResponse => {
+  return Object.entries(properties).reduce(
+    (acc, [fieldName, unparsedField]) => {
+      const fieldPath = [...path, fieldName].join('.');
+      const { value: parsedField, parametersRemoved, meta } = parseField(unparsedField);
+
+      if (parsedField === undefined) {
+        // Field has been stripped out because it was invalid
+        acc.errors.push({ code: 'ERR_FIELD', fieldPath });
+      } else {
+        if (meta!.hasChildFields || meta!.hasMultiFields) {
+          // Recursively parse all the possible children ("properties" or "fields" for multi-fields)
+          const parsedChildren = parseFields(parsedField[meta!.childFieldsName!], [
+            ...path,
+            fieldName,
+          ]);
+          parsedField[meta!.childFieldsName!] = parsedChildren.value;
+
+          /**
+           * If the children parsed have any error we concatenate them in our accumulator.
+           */
+          if (parsedChildren.errors) {
+            acc.errors = [...acc.errors, ...parsedChildren.errors];
+          }
+        }
+
+        acc.value[fieldName] = parsedField;
+
+        if (Boolean(parametersRemoved.length)) {
+          acc.errors = [
+            ...acc.errors,
+            ...parametersRemoved.map(paramName => ({
+              code: 'ERR_PARAMETER' as 'ERR_PARAMETER',
+              fieldPath,
+              paramName,
+            })),
+          ];
+        }
+      }
+
+      return acc;
+    },
+    {
+      value: {},
+      errors: [],
+    } as PropertiesValidatorResponse
+  );
+};
+
+/**
+ * Utility function that reads a mappings "properties" object and validate its fields by
+ * - Removing unknown field types
+ * - Removing unknown field parameters or field parameters that don't have the correct format.
+ *
+ * This method does not mutate the original properties object. It returns an object with
+ * the parsed properties and an array of field paths that have been removed.
+ * This allows us to display a warning in the UI and let the user correct the fields that we
+ * are about to remove.
+ *
+ * NOTE: The Joi Schema that we defined for each parameter (in "parameters_definition".tsx)
+ * does not do an exhaustive validation of the parameter value.
+ * It's main purpose is to prevent the UI from blowing up.
+ *
+ * @param properties A mappings "properties" object
+ */
+export const validateProperties = (properties = {}): PropertiesValidatorResponse => {
```

Comment:
This looks unused too. Removing it will let us remove its test file too.

Reply:
Actually, this one is quite important as it is used by the `validateMappings()` function on L282

Reply:
EDIT: Sorry my bad, `validateMappings()` is not used on this branch, great catch! I also removed it.